pkg/controllers/resources/pods/validatePodSecurity.go

```
@@ -0,0 +1,156 @@
+			pod.Status.Message = result.Result.Message
+			pod.Status.Reason = string(result.Result.Reason)
+			_ = ctx.VirtualClient.Status().Update(ctx.Context, pod)
+			s.EventRecorder().Eventf(pod, "Warning", "SyncError", `Pod %s is forbidden: violates PodSecurity "%s:latest"`, pod.Name, s.podSecurityStandard)
+		}
+		return result.Allowed, nil
+	}
+	return false, err
+}
+
+func (s *podSyncer) validatePodSecurityStandards(ctx context.Context, pod *corev1.Pod) (*admissionv1.AdmissionResponse, error) {
+	if _, ok := podsecurity.PodSecurityExemptNamespaces[pod.Namespace]; ok {
+		return &admissionv1.AdmissionResponse{
+			Allowed: true,
+		}, nil
+	}
+
+	version := api.LatestVersion()
+	podSecurityDefaults := admissionapi.PodSecurityDefaults{
+		Enforce:        s.podSecurityStandard,
+		EnforceVersion: version.String(),
+		Audit:          s.podSecurityStandard,
+		AuditVersion:   version.String(),
+		Warn:           s.podSecurityStandard,
+		WarnVersion:    version.String(),
+	}
+
+	namespaceLister := &namespaceLister{
+		client: s.virtualClusterClient,
+		ctx:    ctx,
+	}
+
+	evaluator, err := policy.NewEvaluator(policy.DefaultChecks())
+	if err != nil {
+		return nil, fmt.Errorf("could not create PodSecurityRegistry: %w", err)
+	}
+
+	adm := &admission.Admission{
+		Configuration: &admissionapi.PodSecurityConfiguration{
+			Defaults: podSecurityDefaults,
+		},
+		Evaluator:        evaluator,
+		Metrics:          &FakeMetricsRecorder{},
+		PodSpecExtractor: admission.DefaultPodSpecExtractor{},
+		PodLister:        admission.PodListerFromClient(s.virtualClusterClient),
+		NamespaceGetter:  admission.NamespaceGetterFromListerAndClient(namespaceLister, s.virtualClusterClient),
+	}
+
+	if err := adm.CompleteConfiguration(); err != nil {
+		return nil, fmt.Errorf("configuration error: %w", err)
+	}
+
+	if err := adm.ValidateConfiguration(); err != nil {
+		return nil, fmt.Errorf("invalid configuration: %w", err)
+	}
+
+	nsPolicy, _ := adm.PolicyToEvaluate(map[string]string{})
```

can we safely ignore the error here?

---

I think yes. Anyway, we're using an empty map there.